File: ws/villagomez/u2/ws21zooWithInheritance/ws21Zoo/src/main/java/ec/espe/ws21zoo/model/Lion.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.espe.ws21zoo.model;

import java.util.Date;

/**
 *
 * @author Domenica Villagomez, CommitCrew, DCCO-ESPE
 */
public class Lion extends Feline{
    
    private boolean mane;

    public Lion(boolean mane, int id, String description, Date bornOnDate, char gender, int numberOfBones, Cage cage) {
        super(id, description, bornOnDate, gender, cage, numberOfBones);
        this.mane = mane;
    }

    
    public void feed(int quantity, Food food){
        System.out.println("Feeding the lion with " + quantity + " of " + food);
    } 

    /**
     * @return the mane
     */
    public boolean isMane() {
        return mane;
    }

    /**
     * @param mane the mane to set
     */
    public void setMane(boolean mane) {
        this.mane = mane;
    }

    public void Feed(int i) {
        throw new UnsupportedOperationException("Not supported yet."); // Generated from nbfs://nbhost/SystemFileSystem/Templates/Classes/Code/GeneratedMethodBody
    }
    
    
    
}

File: hw/viteri/u2/HW18ZooInPython/Zoo/model/Lion.py
from datetime import date
from model.Feline import Feline
from model.Cage import Cage
from model.Food import Food

class Lion(Feline):
    def __init__(self, id, description, born_on_date, gender, cage, number_of_bones, vertebrate, mane=False):
        super().__init__(id, description, born_on_date, gender, cage, number_of_bones, vertebrate)
        self._mane = mane

    def __str__(self):
        return f"Lion{{mane={self._mane}, {super().__str__()}}}"

    @property
    def mane(self):
        return self._mane

    @mane.setter
    def mane(self, value):
        self._mane = value

    def feed(self, food_amount):
        print(f"Feeding the lion with {food_amount} mice")

    def feed_with_food(self, food):
        print(f"Feeding the lion with {food}")

# For compatibility with Java code, you may use a separate feed method
# for handling the Food object if you wish to mimic Java overloading.
